### PR TITLE
Di 21009 array fields

### PR DESCRIPTION
**Pull request by michaelatias**

### CODE REVIEW

File: client-side/src/app/generic-viewer/generic-viewer.component.ts [hunk 1]
```diff
@@ -130,6 +130,28 @@ export class GenericViewerComponent implements OnInit {
...
+        return this.reformatItem(item, resourceFields)
+      })
+    }
+    reformatItem(item, resourceFields){
+      Object.keys(item).forEach(fieldID => {
+        item[fieldID] = resourceFields[fieldID] != undefined? this.getFixedField(item, fieldID, resourceFields): item[fieldID]
+      })
+    }
+    getFixedField(item, fieldID, resourceFields){
+      if(resourceFields[fieldID].Type == "Array"){
```
Comment (roiaharon-pepperi) [inline client-side/src/app/generic-viewer/generic-viewer.component.ts:144]: what about 'Resource' fields not of type Array?

File: client-side/src/app/type-map.ts [hunk 2]
```diff
@@ -15,16 +15,9 @@ export class TypeMap{
         this.map['Bool'] = "Boolean"
         this.map['DateTime'] = "DateAndTime"
         this.map["Resource"] = "Button"
-        return this
+        this.map["Array"] = "TextBox"
```
Comment (roiaharon-pepperi) [inline client-side/src/app/type-map.ts:18]: why 'TextBox' and not 'TextArea'? 
in addition, you deleted the 'return' statement, is it on purpose?

File: client-side/src/app/type-map.ts [hunk 3]
```diff
@@ -15,16 +15,9 @@ export class TypeMap{
...
-        if(key == 'Array'){
-            return optionalValues?.length > 0? "MultiTickBox" : "TextArea"
-        }
-        else if(optionalValues.length > 0){
-            return "ComboBox"
-        }
-        const result = this.map[key]
-        return result; 
+    get(key:string): DataViewFieldType{
+       return this.map[key]
```
Comment (roiaharon-pepperi) [inline client-side/src/app/type-map.ts:21]: Why did you removed this code? It's still relevant. I didn't see that you moved it somewhere else

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: client-side/src/app/type-map.ts (hunk 2 above)
Comment (michaelatias) [reply client-side/src/app/type-map.ts:18]: yes its on purpose. and Eyal want it to be a string as TextBox

File: client-side/src/app/type-map.ts (hunk 3 above)
Comment (gilfoyle12) [reply client-side/src/app/type-map.ts:21]: Eyal changed the reqiurements, arrays should be shown as textBox and the value will be a string that represent the array(thas why in the init function type array convert to TextBox)

File: client-side/src/app/generic-viewer/generic-viewer.component.ts (hunk 1 above)
Comment (gilfoyle12) [reply client-side/src/app/generic-viewer/generic-viewer.component.ts:144]: i will support that in the future, right now thats not in this JIRA

File: client-side/src/app/type-map.ts (hunk 3 above)
Comment (michaelatias) [reply client-side/src/app/type-map.ts:21]: Eyal changed the reqiurements, arrays should be shown as textBox and the value will be a string that represent the array(thas why in the init function type array convert to TextBox)

File: client-side/src/app/generic-viewer/generic-viewer.component.ts (hunk 1 above)
Comment (michaelatias) [reply client-side/src/app/generic-viewer/generic-viewer.component.ts:144]: i will support that in the future, right now thats not in this JIRA

**Pull request merged**